### PR TITLE
RavenDB-15772 - Have a dedicated raft command to change the client/st…

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-15772/Have-a-dedicated-raft-command-to-change-the-clientstudiosettings-configuration-from-AdminConfigurationHandler

### Additional description

Have a dedicated raft command to change the client/studio/settings configuration from AdminConfigurationHandler.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Admin/AdminConfigurationHandler.cs [hunk 1]
```diff
@@ -5,17 +5,20 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features.Authentication;
+using MySqlX.XDevAPI;
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Admin/AdminConfigurationHandler.cs]: huh?

Comment (shaharhikri) [reply src/Raven.Server/Documents/Handlers/Admin/AdminConfigurationHandler.cs]: Removed

File: src/Raven.Server/ServerWide/ClusterCommandsVersionManager.cs [hunk 2]
```diff
@@ -153,6 +153,9 @@ public static class ClusterCommandsVersionManager
             [nameof(UpdateQueueEtlCommand)] = 54_000,
             [nameof(PutQueueConnectionStringCommand)] = 54_000,
             [nameof(RemoveQueueConnectionStringCommand)] = 54_000,
+            [nameof(PutDatabaseStudioConfigurationCommand)] = 54_000,
+            [nameof(PutDatabaseSettingsCommand)] = 54_000,
+            [nameof(PutDatabaseClientConfigurationCommand)] = 54_000,
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/ClusterCommandsVersionManager.cs]: need to bump those to 54_001

File: src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amqp.Framing;
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs]: what is that?

File: src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+using Sparrow;
+using Sparrow.Json.Parsing;
+using Index = Raven.Server.Documents.Indexes.Index;
+
+namespace Raven.Server.ServerWide.Commands
+{
+    public class PutDatabaseClientConfigurationCommand : UpdateDatabaseCommand
+    {
+
+        public ClientConfiguration Client;
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs]: let's rename `Client` to `Configuration`

File: src/Raven.Server/ServerWide/Commands/PutDatabaseSettingsCommand.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amqp.Framing;
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/Commands/PutDatabaseSettingsCommand.cs]: same

File: src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+using Raven.Server.Utils;
+using Sparrow;
+using Sparrow.Json.Parsing;
+using Index = Raven.Server.Documents.Indexes.Index;
+
+namespace Raven.Server.ServerWide.Commands
+{
+    public class PutDatabaseStudioConfigurationCommand : UpdateDatabaseCommand
+    {
+        public StudioConfiguration Studio;
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs]: rename to `Configuration`

File: src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Amqp.Framing;
```
Comment (karmeli87) [inline src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs]: same

File: test/SlowTests/Issues/RavenDB-15772.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+using Raven.Client.Documents.Operations.Configuration;
+using Raven.Client.ServerWide.Operations;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_15772 : RavenTestBase
+    {
+        public RavenDB_15772(ITestOutputHelper output) : base(output)
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-15772.cs:44]: let's create a test for each of the commands

File: test/SlowTests/Issues/RavenDB-15772.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            var updateClientConfig = store.Maintenance.SendAsync(
+                new PutClientConfigurationOperation(
+                    new ClientConfiguration
+                    {
+                        MaxNumberOfRequestsPerSession = 30
+                    }));
+
+            var updateTopology = store.Maintenance.Server.SendAsync(new ReorderDatabaseMembersOperation(store.Database, fixedOrder, true));
+
+            await Task.WhenAll(updateClientConfig, updateTopology);
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-15772.cs:63]: let's read the configuration and assert it was indeed changed

File: src/Raven.Server/ServerWide/ClusterCommandsVersionManager.cs (hunk 2 above)
Comment (shaharhikri) [reply src/Raven.Server/ServerWide/ClusterCommandsVersionManager.cs]: Done

File: src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs (hunk 3 above)
Comment (shaharhikri) [reply src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs]: Done

File: src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs (hunk 4 above)
Comment (shaharhikri) [reply src/Raven.Server/ServerWide/Commands/PutDatabaseClientConfigurationCommand.cs]: Done

File: src/Raven.Server/ServerWide/Commands/PutDatabaseSettingsCommand.cs (hunk 5 above)
Comment (shaharhikri) [reply src/Raven.Server/ServerWide/Commands/PutDatabaseSettingsCommand.cs]: Done

File: src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs (hunk 6 above)
Comment (shaharhikri) [reply src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs]: Done

File: src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs (hunk 7 above)
Comment (shaharhikri) [reply src/Raven.Server/ServerWide/Commands/PutDatabaseStudioConfigurationCommand.cs]: Done

File: test/SlowTests/Issues/RavenDB-15772.cs (hunk 8 above)
Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-15772.cs:44]: Done

File: test/SlowTests/Issues/RavenDB-15772.cs (hunk 9 above)
Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-15772.cs:63]: Done

**Review by karmeli87 — APPROVED** [review]

**Comment by ppekrol** [comment]
@shaharhikri Can you check failing test please?

File: test/SlowTests/Issues/RavenDB_15772_TypeConverter_Fail_To_Convert_Char.cs [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+namespace SlowTests.Issues
+{
+    public class RavenDB_15772_TypeConverter_Fail_To_Convert_Char : ClusterTestBase
+    {
+        public RavenDB_15772_TypeConverter_Fail_To_Convert_Char(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void Fail_To_Convert_Char_To_Blittable()
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB_15772_TypeConverter_Fail_To_Convert_Char.cs]: let's rename this `Can_Convert_Char_To_Blittable`

File: test/SlowTests/Issues/RavenDB_15772_TypeConverter_Fail_To_Convert_Char.cs [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_15772_TypeConverter_Fail_To_Convert_Char : ClusterTestBase
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB_15772_TypeConverter_Fail_To_Convert_Char.cs:22]: cc @ppekrol should we backport this fix to 5.2 ?

File: test/SlowTests/Issues/RavenDB_15772_TypeConverter_Fail_To_Convert_Char.cs (hunk 10 above)
Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB_15772_TypeConverter_Fail_To_Convert_Char.cs]: Done

**Review by ppekrol — APPROVED** [review]

**Comment by shaharhikri** [comment]
I wrapped the expected strings by ```SourceCodeBeautifier.FormatIndex(map).Expression``` in ```IndexMerging``` tests but I rebased after those tests were fixed and merged..

**Pull request merged**